### ISSUE TITLE
How to find the clients connected with MAVSDK Server?

### ISSUE THREAD

**Issue by MuhammadBilal1** [JonasVautherin/px4-gazebo-headless#13]
I am running the [Gazebo Headless Simulator](https://github.com/jonasvautherin/px4-gazebo-headless#px4-gazebo-simulator-headless) with the below command.

`docker run --rm -it jonasvautherin/px4-gazebo-headless:1.10.1`

After successfully running the simulator, I tried to ran the [example](https://github.com/mavlink/MAVSDK-Python/blob/master/examples/mission.py). 
`python mission.py` This example ran successfully.

But when I try to ran the mission example from my own application (basically a Django application with Celery for running MAVSD in background) If I get any problem while running the example, after the error I am unable to connect with the Drone (Gazebo Headless Simulator). I tried to stop the simulator and started again, but still not working.

1- How to check is there any client connected with MAVLINK?
2- If an error came while running the example and program failed without completing the "Landing", the problem didn't able to connect with the drone after the error.

![image](https://user-images.githubusercontent.com/7131058/87951155-a5f4bc80-cac1-11ea-94cd-df4f4110b744.png)

**Comment by MuhammadBilal1** [comment]
@julianoes @JonasVautherin  Can you please help me with the above problem.

**Comment by JonasVautherin** [comment]
I'm not sure what kind of error you are referring to. You mean that `mission.py` crashes?

**Comment by MuhammadBilal1** [comment]
Error is appearing while executing the mission.py from my Django application via Celery.

Error is coming while uploading the mission to the connected drone. Now, after getting this error, I can't connect to the drone again until I restart my Operating System, or I close all my terminal windows, which are running Django/Celery/Gazebo HeadLess Simulator.

**Problems:**
1- Getting the `Operation Timeout error` at `upload_mission()` as shown in the below image while running the mission.py.
2- After getting the operation timeout error, mission.py remains stuck at `Waiting for Drone to connect...`.

![image](https://user-images.githubusercontent.com/7131058/87959031-bad64d80-cacb-11ea-814e-d5f782416554.png)

**Comment by MuhammadBilal1** [comment]
Somehow I managed to get the list of ports that were in use when I am unable to connect with the Drone. 
Please see the attached snapshot for details.
**I need help in resolving the Operation Timeout error of mission.py**


![Ubuntu18 04 04_64-bit-2020-07-21-16-56-52](https://user-images.githubusercontent.com/7131058/88053090-b0bc5980-cb74-11ea-969a-b42bfd991a1d.png)

**Comment by JonasVautherin** [comment]
Maybe you should try to run `mavsdk_server` manually, with:

```
./mavsdk_server -p 50051 udp://:14540
```

And in Python, you need to connect to it by changing the `System()` line:

```
drone = System(mavsdk_server_address='localhost', port=50051)
```

**Comment by MuhammadBilal1** [comment]
I am unable to start the `mavsdk_server` manually, can you please tell me from where I need to run this command?
Please see the attached snapshot.

![image](https://user-images.githubusercontent.com/7131058/88089763-ff361c00-cba5-11ea-9107-a3e714120e98.png)

**Comment by julianoes** [comment]
You can look for it:
```
find . -name mavsdk_server
```

For me, the one installed using pip is here:
```
~/.local/lib/python3.8/site-packages/mavsdk/bin/mavsdk_server
```

**Comment by MuhammadBilal1** [comment]
@julianoes  Thank you for your support.
I installed the mavsdk using the pip, and my installation is also available at the same place. I went to that directory and started the mavsdk_server manually. The server started successfully.

**My issue is resolved and I am not getting the `Operation timeout` error anymore while running the `mission.py` example.**

I started the things in the following order, and everything worked fine.

1- Started mavsdk_server manually from your given command. Ran successfully.
2- Started the Gazebo Headless simulator. Ran successfully.
3- Started by application (Django). Ran successfully.
4- Started Celery. Ran successfully.
5- Triggered the method which was calling the `mission.py`. It also ran successfully.

![image](https://user-images.githubusercontent.com/7131058/88180649-1d078d80-cc47-11ea-9093-6d24b701ec29.png)
